### DISCUSSION TITLE
Extracting Embeddings

### DISCUSSION THREAD

**Discussion by aivyon** [PaddlePaddle/PaddleOCR#12204] [Q&A]
I am currently working on performing inference using PaddleOCR for text recognition tasks. 
During the inference process, I'm able to to recognize text. 
However, I'm interested in extracting the embeddings just before the softmax argmax operation. 
These embeddings would provide valuable information for further analysis and processing. 
Could you please guide me on how to access these embeddings in the PaddleOCR framework?
I tried to use the following code :
import paddleocr.pproc.inference.wrapper as wrapper
predictor = wrapper.Predictor()
embeddings = predictor.run(imglist)

However, this resulted in the following error:
ImportError: generic_type: type "VarBase" is already registered!     (fluid/core.py",line 197, in <module> from. import libpaddle

 Any insights or code examples would be greatly appreciated. Thank you!

**Comment by aivyon** [comment]
anyone?:)

**Comment by aditya-ihx** [comment]
@divastar Did you find a way to extract the embeddings ?

**Comment by aivyon** [comment]
[aditya-ihx](https://github.com/aditya-ihx)
No.
tried:
predictor = wrapper.Predictor()
embeddings = predictor.run(imglist)

but it is failing in the framework.core.py
so I tried to mock the core.py  
because I don't really need it for the embeddings:

import unittest.mock
paddle.fluid.core = unittest.mock.MagicMock()

but getting circular imports problems the system breaks

**Comment by aditya-ihx** [comment]
I am trying to accomplish the same goal. As of now I have nothing, will update you over the weekend if there is any progress 
@divastar

**Comment by aivyon** [comment]
> I am trying to accomplish the same goal. As of now I have nothing, will update you over the weekend if there is any progress @divastar

Great! Thank you  🙏
@aditya-ihx
